compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -69,8 +69,34 @@ import SdkVersion
 --   ledger. Based on the DAML-LF we generate dummy interface files
 --   and then remap references to those dummy packages to the original DAML-LF
 --   package id.
-createProjectPackageDb :: NormalizedFilePath -> Options -> PackageSdkVersion -> [FilePath] -> [FilePath] -> IO ()
+
+calcUnitsFromDeps :: [FilePath] -> IO [UnitId]
+calcUnitsFromDeps deps = do
+    let (fpDars, fpDalfs) = partition ((== ".dar") . takeExtension) deps
+    dars <- mapM extractDar fpDars
```

Comment:
This is a bit wasteful. We are now extracting each DAR twice, once here and then once at https://github.com/digital-asset/daml/blob/36e188cac84ddb0f4259cd077a7bb6b4fd466311/compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs#L115. We are also decoding everything twice, once here, and once at https://github.com/digital-asset/daml/blob/36e188cac84ddb0f4259cd077a7bb6b4fd466311/compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs#L130. Can we factor this such that we only have to do this once?

Reply:
as discussed.
our new plan to relocate the `calUnitsForDeps` code means this sharing wont be possible now